test/e2e/main_test.go

```diff
@@ -142,3 +143,29 @@ func TestMemoryUsageRecordingRule(t *testing.T) {
 		"count(namespace:container_memory_usage_bytes:sum)",
 	)
 }
+
+// This makes sure that the priority class we create is
+// present and lower than the system priority classes. Example:
+// openshift-user-critical   1000000000   false            66m
+// system-cluster-critical   2000000000   false            114m
+// system-node-critical      2000001000   false            114m
+func TestToEnsureUserPriorityClassIsPresentAndLower(t *testing.T) {
+	// Get system priority class values.
+	systemClusterPriorityClass, err := f.SchedulingClient.PriorityClasses().Get(context.TODO(), "system-cluster-critical", v1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	systemNodePriorityClass, err := f.SchedulingClient.PriorityClasses().Get(context.TODO(), "system-node-critical", v1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Get our user priority class value.
+	userPriorityClass, err := f.SchedulingClient.PriorityClasses().Get(context.TODO(), "openshift-user-critical", v1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Ensure our is a lower value than the system class' ones.
+	if userPriorityClass.Value >= systemClusterPriorityClass.Value && userPriorityClass.Value >= systemNodePriorityClass.Value {
```

Comment:
logical question: shouldn't this be an or `||`? This would pass if i.e. the user priority is higher than system cluster priority but lower than the system node priority, aka "in between".

Reply:
Indeed good catch!